### ISSUE TITLE
v3 beta: cannot find function `max_concurrent_ssl_connect` in crate `actix_tls`

### ISSUE THREAD

**Issue by robjtede** [actix/actix-web#1657]
The semver rules in Cargo are weird when it comes to transitioning out of pre-releases. Add the following to your Cargo.toml until next version is released:

```toml
actix-tls = "=2.0.0-alpha.2"
```

**Comment by therealprof** [comment]
That's not the only breaking change... Also the Ssl -> Tls renaming creates some woes...

**Comment by Neopallium** [comment]
would it be possible to use type aliases for the old type names and deprecated methods for the renamed methods?  Give users a little bit more time to convert code.

**Comment by robjtede** [comment]
Did consider it, but there's breaking changes in the -tls release anyway and mostly just easy renames. It's also not possible to alias a single enum variant otherwise I'd have gone through with it.

We'll get this remedied ASAP.

**Comment by guilt** [comment]
Would suggest doing a new release because the `beta3` is already breaking for me :( Currently I'm hotpatching and renaming that function to `max_concurrent_tls_connect` to work around

**Comment by robjtede** [comment]
beta 4 is released

**Comment by therealprof** [comment]
Now `actix-web-actors` is broken and needs a release... 😅 

```
    Checking actix-web-actors v3.0.0-beta.1
error[E0599]: no method named `encode` found for struct `actix_http::ws::Codec` in the current scope
   --> /Users/egger/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-actors-3.0.0-beta.1/src/ws.rs:438:30
    |
438 |                 this.encoder.encode(msg, &mut this.buf)?;
    |                              ^^^^^^ method not found in `actix_http::ws::Codec`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use tokio_util::codec::encoder::Encoder;`
```

**Comment by jplatte** [comment]
Why are the beta versions of the actix crates that themselves depend on other betas not using exact version requirements?

**Comment by robjtede** [comment]
It's a burden to do that across more than 15 crates since it assumes that every beta release is breaking from the last. Sorry for any inconvenience but beta releases would be slower if we pinned everything.

With effectively caret beta versions you are able to pin anything that breaks.

**Comment by therealprof** [comment]
Yeah, it's a bit unfortunate that this means that beta testers basically constantly have to wiggle pins in order to both test the latest versions and not getting broken by surprise the next day. It's a bit of whack-a-mole to stay up-to-date, I have to fiddle with the dependencies every other day at the moment...

Unfortunately we crucially depend on the beta versions so we don't have a choice... My days are going to be much better once the next stable lands...

**Comment by robjtede** [comment]
Thanks for understanding. I feel the pain too with personal and work projects. Won't be much longer now.
